app/src/main/java/com/hedvig/app/feature/offer/OfferRepository.kt

```diff
@@ -45,10 +47,16 @@ class OfferRepository(
             .toEither { ErrorMessage(it) }
             .bind()
 
-        ensureNotNull(result.quoteCart.fragments.quoteCartFragment.bundle) {
+        val quoteCartFragment = result.quoteCart.fragments.quoteCartFragment
+        val bundle = quoteCartFragment.bundle
+        ensureNotNull(bundle) {
             ErrorMessage("No quotes in offer, please try again")
         }
+        val receivedQuoteIds = bundle.possibleVariations
+            .flatMap { variation -> variation.bundle.fragments.quoteBundleFragment.quotes }
+            .map { quote -> quote.id }
+        hAnalytics.receivedQuotes(receivedQuoteIds)
```

Comment:
Could we have logged this higher up, in the viewmodel perhaps? To avoid sending multiple analytics events.

Reply:
Hmm I've thought of that as well, but that leaves us with only sending the event sometimes when we're doing this network request, which means we have to manually think each time we request for the quotes if we want to log it this time or not. And depending on how this evolves it may mean that we have to then add the event in two or more places and in general wonder if we're under-sending it all the time. So it made sense to me to simply send the event when we're doing the network request. And this makes it so that for the analytics team to get a good grip of what is happening to filter out the absolutely equivalent events (same quote cart ID and same exact list of quote IDs in it). If we revisit this in the future we can look into improving it for sure. If this feels like a blocker to you I can definitely try and re-iterate on it a bit.

Reply:
I see, then this implementation is reasonable.